### PR TITLE
Follow posts and share button

### PR DESCRIPTION
**Pull request by shavnita**
Added share button functionality

### CODE REVIEW

File: wp-content/themes/oceanwp-child/assets/css/template-styles.css [hunk 1]
```diff
@@ -172,8 +177,6 @@ hr{
     color:#275f87
 }
 
-
-
 .blog-sub-header{
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/assets/css/template-styles.css:180]: Can we go with `subheader` and not `sub-header` ?

File: wp-content/themes/oceanwp-child/assets/css/template-styles.css [hunk 2]
```diff
@@ -273,4 +276,10 @@ hr{
 .ini-supporters{
     color:#DC9C05;
     margin-bottom:0px
+}
+
+.ess-list{
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/assets/css/template-styles.css:281]: What is `ess`?

File: wp-content/themes/oceanwp-child/functions.php [hunk 3]
```diff
@@ -259,4 +260,70 @@ function add_related_news_to_initiative_pages() {
...
+		wp_redirect('http://localhost/ggrc_website/'); 
+		
+		exit;
+	}
+	else{
+	
+		$current_user_id = get_current_user_id();
+		$postid = $_POST['postid'];
+
+		//echo $current_user_id;
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/functions.php:282]: delete comment.
Could you also please reduce the amount of newlines?

File: wp-content/themes/oceanwp-child/functions.php [hunk 4]
```diff
@@ -259,4 +260,70 @@ function add_related_news_to_initiative_pages() {
...
+			
+		$table_name = $wpdb->prefix . 'follow_posts';     
+		 
+		$wpdb->query($wpdb->prepare("UPDATE $table_name SET isFollowing = '0' WHERE userID = '$current_user_id' and postID = '$postid'"));		
+		
+	}
+
+}
+
+//add_action( 'check_follow', 'check_following' );
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/functions.php:318]: uncomment or remove

File: wp-content/themes/oceanwp-child/functions.php [hunk 5]
```diff
@@ -259,4 +260,70 @@ function add_related_news_to_initiative_pages() {
...
+
+function check_following() {
+	global $wpdb;
+
+	$current_user_id = get_current_user_id();
+	$postid= get_the_ID();
+	$posts = $wpdb->get_results("SELECT * FROM ggrc_follow_posts WHERE `userID` = '$current_user_id' and `postID` = '$postid' and `isFollowing` = 1");
+	
+	//var_dump($posts);
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/functions.php:327]: Uncomment or delete

File: wp-content/themes/oceanwp-child/functions.php [hunk 6]
```diff
@@ -259,4 +260,70 @@ function add_related_news_to_initiative_pages() {
...
+		 
+		$wpdb->query($wpdb->prepare("UPDATE $table_name SET isFollowing = '0' WHERE userID = '$current_user_id' and postID = '$postid'"));		
+		
+	}
+
+}
+
+//add_action( 'check_follow', 'check_following' );
+
+function check_following() {
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/functions.php:320]: I'm having a hard time understanding what this function is doing.
Could you please change the name to be meaningful. Something like: `countInitiativeFollowers` maybe?

File: wp-content/themes/oceanwp-child/single-blog_post.php [hunk 7]
```diff
@@ -17,32 +17,32 @@
...
+
+		.wpulike-heart .count-box{
+			color: #fff !important
+		}
+
+		.ess-all-networks-button, #ess-main-wrapper .ess-total-share{
+			background: linear-gradient(80.63deg, #0B4F6D 26.04%, #0F7AA9 99.31%) !important;
+			box-shadow: 2px 3px 9px rgba(0, 0, 0, 0.2) !important;
+			border-radius: 30px !important;
+		}
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/single-blog_post.php:36]: I think I've created a `child-plugin` for this plugin. Maybe we should add this style there?

File: wp-content/themes/oceanwp-child/single-blog_post.php [hunk 8]
```diff
@@ -17,32 +17,32 @@
...
+
+		@media only screen and (min-width: 960px){
+			.content-area, .content-left-sidebar .content-area{
+				width:75%
+			}
+			.widget-area.sidebar-primary {
+				width: 25%;
+			}
+		}
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/single-blog_post.php:45]: Is this true for all pages? Or just the blog?

File: wp-content/themes/oceanwp-child/single-initiative.php [hunk 9]
```diff
@@ -317,18 +344,51 @@
...
+
+			jQuery.ajax({
+				url: ajaxurl, // this will point to admin-ajax.php
+				type: 'POST',
+				data: data,
+				success: function (response) {
+					console.log(response); 
+					$("#unfollow").html("follow initiative");
+					$('#unfollow').attr('id','follow'); 
+					//document.getElementById("follow").innerHTML="<button id=\"unfollow\" class=\"template-btn\">unfollow</button>";
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/single-initiative.php:387]: Uncomment or delete

File: wp-content/themes/oceanwp-child/single-initiative.php [hunk 10]
```diff
@@ -317,18 +344,51 @@
...
+				data: data,
+				success: function (response) {
+					console.log(response); 
+					$("#unfollow").html("follow initiative");
+					$('#unfollow').attr('id','follow'); 
+					//document.getElementById("follow").innerHTML="<button id=\"unfollow\" class=\"template-btn\">unfollow</button>";               
+				}
+			});
 		});
-	});
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/single-initiative.php:330]: Maybe all this should go out to an asset JS file?

File: wp-content/themes/oceanwp-child/social_share/share.php [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+if ( 'no' === get_option( 'easy_social_sharing_inline_enable_share_counts' ) ) {
+	$visbility_class[] = 'ess-no-share-counts';
+}
+
+if ( 'no' === get_option( 'easy_social_sharing_inline_enable_networks_label' ) ) {
+	$visbility_class[] = 'ess-no-network-label';
+}
+
+$social_networks = ess_get_core_supported_social_networks();
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/social_share/share.php:38]: No closing `?>`

File: wp-content/themes/oceanwp-child/social_share/share.php [hunk 12]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+global $post;
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/social_share/share.php:3]: Please wrap this part in a function with a meaningful name.
Also, the filename is not descriptive to (what I think) this is doing. Also change the file's name.

File: wp-content/themes/oceanwp-child/assets/css/template-styles.css (hunk 1 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/assets/css/template-styles.css:180]: Done

File: wp-content/themes/oceanwp-child/assets/css/template-styles.css (hunk 2 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/assets/css/template-styles.css:281]: easy social sharing, I've extended this css in initiatives share button. They're using ess prefix for their css classes.

File: wp-content/themes/oceanwp-child/single-blog_post.php (hunk 7 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/single-blog_post.php:36]: In the layout page itself or you want me to create separate style sheet for it?

File: wp-content/themes/oceanwp-child/single-blog_post.php (hunk 8 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/single-blog_post.php:45]: Just blog that's why I've put it there.

File: wp-content/themes/oceanwp-child/functions.php (hunk 3 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/functions.php:282]: Deleted

File: wp-content/themes/oceanwp-child/functions.php (hunk 6 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/functions.php:320]: Function is checking if a user is following a certain initiative post. Changed the name to checkIsFollowing();

File: wp-content/themes/oceanwp-child/social_share/share.php (hunk 12 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/social_share/share.php:3]: Removed the file, I don't really need it.

File: wp-content/themes/oceanwp-child/single-blog_post.php (hunk 7 above)
Comment (YonatanKra) [reply wp-content/themes/oceanwp-child/single-blog_post.php:36]: Let's start with the layout page. If qe see we need more tweaks and variants we can create a file.

File: wp-content/themes/oceanwp-child/assets/css/template-styles.css [hunk 13]
```diff
@@ -273,4 +276,10 @@ hr{
 .ini-supporters{
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/assets/css/template-styles.css:276]: can you please change this to `initiative-supporters`?

File: wp-content/themes/oceanwp-child/functions.php [hunk 14]
```diff
@@ -252,11 +243,55 @@ function add_related_news_to_initiative_pages() {
...
+
+function checkIsFollowing() {
+	global $wpdb;
+
+	$current_user_id = get_current_user_id();
+	$postid= get_the_ID();
+	$posts = $wpdb->get_results("SELECT DISTINCT userID, postID FROM ggrc_follow_posts WHERE `userID` = '$current_user_id' and `postID` = '$postid' and `isFollowing` = 1");
+
+	return count($posts);
 }
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/functions.php:297]: Probably needs a closing `?>`?

File: wp-content/themes/oceanwp-child/functions.php [hunk 15]
```diff
@@ -252,11 +243,55 @@ function add_related_news_to_initiative_pages() {
...
+
+	$current_user_id = get_current_user_id();
+	$postid = $_POST['postid'];			
+	$table_name = $wpdb->prefix . 'follow_posts';     
+		
+	$wpdb->query($wpdb->prepare("UPDATE $table_name SET isFollowing = '0' WHERE userID = '$current_user_id' and postID = '$postid'"));		
+		
+}
+
+function checkIsFollowing() {
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/functions.php:289]: Just a suggestion
```suggestion
function checkIsUserFollowing() {
```

File: wp-content/themes/oceanwp-child/single-blog_post.php (hunk 8 above)
Comment (YonatanKra) [reply wp-content/themes/oceanwp-child/single-blog_post.php:45]: Isn't it true also for initiatives page?

File: wp-content/themes/oceanwp-child/single-initiative.php [hunk 16]
```diff
@@ -43,8 +54,7 @@
 			<main id="main" class="site-main clr"<?php oceanwp_schema_markup( 'main' ); ?> role="main">
 
 				<?php //do_action( 'ocean_page_header' ); ?>
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/single-initiative.php:56]: Delete line?

File: wp-content/themes/oceanwp-child/single-initiative.php [hunk 17]
```diff
@@ -151,6 +178,7 @@
 						the_post(); 
 
 							$the_post_id = get_the_ID();
+							//echo $the_post_id;
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/single-initiative.php:181]: Delete line?

File: wp-content/themes/oceanwp-child/functions.php [hunk 18]
```diff
@@ -252,11 +243,55 @@ function add_related_news_to_initiative_pages() {
...
+	global $wpdb;
+
+	$current_user_id = get_current_user_id();
+	$postid= get_the_ID();
+	$posts = $wpdb->get_results("SELECT DISTINCT userID, postID FROM ggrc_follow_posts WHERE `userID` = '$current_user_id' and `postID` = '$postid' and `isFollowing` = 1");
+
+	return count($posts);
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/functions.php:296]: If this function checks if user is following, why do we count?
Would this work the same and be clearer?
```suggestion
	return isEmpty($posts);
```
The you can just do this in the template:
`<?php if (isCurrentUserFollowing()) { ?>`

File: wp-content/themes/oceanwp-child/single-blog_post.php (hunk 7 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/single-blog_post.php:36]: I have made changes to this one but it seems to be not using the child plugin as the changes didn't take effect, so I had to make the changes in the ess plugin itself. That worked.

File: wp-content/themes/oceanwp-child/single-blog_post.php (hunk 8 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/single-blog_post.php:45]: No, I'm using bootstrap grid there.

File: wp-content/themes/oceanwp-child/assets/css/template-styles.css (hunk 13 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/assets/css/template-styles.css:276]: Done

File: wp-content/themes/oceanwp-child/functions.php (hunk 15 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/functions.php:289]: Done

File: wp-content/themes/oceanwp-child/single-blog_post.php (hunk 7 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/single-blog_post.php:36]: Okay reverting changes for this as I'm using the share button in initiatives as well with different design and adding this style in the layout file over rides my initiative share button styling. Let's leave it in the single blog page.

File: wp-content/themes/oceanwp-child/functions.php (hunk 18 above)
Comment (shavnita) [reply wp-content/themes/oceanwp-child/functions.php:296]: isEmpty function doesn't seem to work so I'm checking like this -> `if (!empty(isCurrentUserFollowing()))`

File: wp-content/themes/oceanwp-child/single-initiative.php [hunk 19]
```diff
@@ -76,18 +86,35 @@
...
-											</button>
+											
+											<?php if ( 'yes' === get_option( 'easy_social_sharing_inline_enable_all_networks' ) ) : ?>
+												<li class="ess-all-networks ess-social-networks ess-list">
+													<div class="ess-social-network-link">
+														
+														<span class="ess-all-networks-button share-initiative" style="color:#0B4F6D !important"><i aria-hidden="true" class="fa fa-share-alt"></i> <b>share initiative</b></span>
+													</div>
+												</li>
+											<?php endif; ?>
```
Comment (YonatanKra) [inline wp-content/themes/oceanwp-child/single-initiative.php:97]: Is this a copy of the plugin's shortcode's template?

File: wp-content/themes/oceanwp-child/functions.php (hunk 18 above)
Comment (YonatanKra) [reply wp-content/themes/oceanwp-child/functions.php:296]: Yes, you are right. `isEmpty` is from another language :)
In PHP it is `empty`. Updated my suggestion

**Pull request closed without merging**